### PR TITLE
Callback to promise

### PR DESCRIPTION
**Pull request by AnnaVerba**

### CODE REVIEW

File: consumer/app.js [hunk 1]
```diff
@@ -1,49 +1,32 @@
...
-            },
-            {
-                noAck: true,
-            },
-        );
+import amqplib from 'amqplib';
+
+connect();
+async function connect() {
+  const amqpServer = 'amqp://localhost:5672';
```
Comment (MaxLogvyniukFive) [inline consumer/app.js]: Move to .env

File: consumer/app.js [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
-                    Buffer.from(msg.content.toString()),
-                );
-            },
-            {
-                noAck: true,
-            },
-        );
+import amqplib from 'amqplib';
+
+connect();
```
Comment (MaxLogvyniukFive) [inline consumer/app.js]: Connect need to use after initialization

File: consumer/app.js [hunk 3]
```diff
@@ -1,49 +1,32 @@
...
+
+    await channel.consume(queue, message, {
+      noAck: true,
     });
-});
-
-module.exports = app;
-
+  } catch (error) {
+    console.log(error);
```
Comment (MaxLogvyniukFive) [inline consumer/app.js]: Add logger

File: consumer/app.js [hunk 4]
```diff
@@ -1,49 +1,32 @@
...
+import amqplib from 'amqplib';
+
+connect();
+async function connect() {
+  const amqpServer = 'amqp://localhost:5672';
+  const connection = await amqplib.connect(amqpServer);
+  const channel = await connection.createChannel();
+  const queue = 'hello';
+  await channel.assertExchange('exchange1', 'topic', {
+    durable: true,
```
Comment (MaxLogvyniukFive) [inline consumer/app.js]: Use variable for true

File: consumer/package-lock.json [hunk 5]
```diff
@@ -0,0 +1,2277 @@
+{
```
Comment (MaxLogvyniukFive) [inline consumer/package-lock.json]: Use yarn instead of npm

File: src/main.ts [hunk 6]
```diff
@@ -22,6 +22,6 @@ async function bootstrap() {
   app.setGlobalPrefix('api');
   app.useGlobalPipes(new ValidationPipe());
 
-  await app.listen(8001);
+  await app.listen(8002);
```
Comment (MaxLogvyniukFive) [inline src/main.ts]: Move to .env

File: src/movie/MessagingService.ts [hunk 7]
```diff
@@ -5,19 +5,18 @@ import { RabbitRPC } from '@golevelup/nestjs-rabbitmq';
...
-        queue: 'hello',
-    })
-    public async rpcHandler(msg: any, amqpMsg: ConsumeMessage) {
-        console.log(
-            'Routing key:' + amqpMsg.fields.routingKey,
-            `\nProducer received :`,
-            amqpMsg.content.toString(),
-        );
+  @RabbitRPC({
+    routingKey: 'replay.#',
```
Comment (MaxLogvyniukFive) [inline src/movie/MessagingService.ts]: Move in variable 'replay.#', Create shared folder and file consts.ts

File: src/movie/MessagingService.ts [hunk 8]
```diff
@@ -5,19 +5,18 @@ import { RabbitRPC } from '@golevelup/nestjs-rabbitmq';
...
+    exchange: 'exchange1',
+    queue: 'hello',
+  })
+  public async rpcHandler(msg: any, amqpMsg: ConsumeMessage) {
+    console.log(
```
Comment (MaxLogvyniukFive) [inline src/movie/MessagingService.ts]: Logger

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]

File: consumer/app.ts [hunk 9]
```diff
@@ -1,32 +1,40 @@
 import amqplib from 'amqplib';
+import logger from '../winstonLogger/Logger';
+import { replayKey } from '../shared/consts';
 
-connect();
+const trueOrFalse = true;
```
Comment (MaxLogvyniukFive) [inline consumer/app.ts]: Name of variable must tell what it contains

File: consumer/dist/consumer/app.js [hunk 10]
```diff
@@ -0,0 +1,82 @@
+'use strict';
```
Comment (MaxLogvyniukFive) [inline consumer/dist/consumer/app.js]: We not pushing dist to git

File: consumer/logs/info.log [hunk 11]
```diff
@@ -0,0 +1,9 @@
+info: Oct-25-2022 13:44:03: 	 [x] Received %s
```
Comment (MaxLogvyniukFive) [inline consumer/logs/info.log]: Add logs to gitignore

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]
Remove dist folder and add it to git ignore

File: consumer/package-lock.json [hunk 12]
```diff
@@ -0,0 +1,7819 @@
...
```
Comment (MaxLogvyniukFive) [inline consumer/package-lock.json]: Please remove package-lock and use yarn to install node modules

File: consumer/app.ts [hunk 13]
```diff
@@ -0,0 +1,59 @@
+import amqplib from 'amqplib';
+import logger from '../winstonLogger/Logger';
+import { replayKey } from '../src/common/constants/consts';
+
+const durableIsTrue = true;
+const hardTestQueue = 'hardTest';
+const amqpServer = 'amqp://localhost:5672';
```
Comment (MaxLogvyniukFive) [inline consumer/app.ts]: Move to .env this url

File: consumer/app.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
+import amqplib from 'amqplib';
+import logger from '../winstonLogger/Logger';
+import { replayKey } from '../src/common/constants/consts';
+
+const durableIsTrue = true;
```
Comment (MaxLogvyniukFive) [inline consumer/app.ts]: Rename to "isDurable"

File: src/main.ts [hunk 15]
```diff
@@ -21,7 +22,7 @@ async function bootstrap() {
   });
   app.setGlobalPrefix('api');
   app.useGlobalPipes(new ValidationPipe());
-
-  await app.listen(8001);
+  // app.use(winston);
```
Comment (MaxLogvyniukFive) [inline src/main.ts]: Why commented?

File: src/modeles/messaging_with_rabbitqm/MessagingService.ts [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+import { ConsumeMessage } from 'amqplib';
+import { RabbitRPC } from '@golevelup/nestjs-rabbitmq';
+import { replayKey } from '../../common/constants/consts';
+import logger from '../../../winstonLogger/Logger';
+
+import { ExchangeType } from '../../common/types';
+const ExchangeMovie: ExchangeType = {
+  routingKey: replayKey,
+  exchange: 'exchange1',
```
Comment (MaxLogvyniukFive) [inline src/modeles/messaging_with_rabbitqm/MessagingService.ts]: Also move to consts

File: src/modeles/messaging_with_rabbitqm/MessagingService.ts [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+  queue: 'hello',
```
Comment (MaxLogvyniukFive) [inline src/modeles/messaging_with_rabbitqm/MessagingService.ts]: Sme here

File: src/modeles/movie/movie.service.ts [hunk 18]
```diff
@@ -49,6 +48,21 @@ export class MovieService {
     return movie;
   }
 
+  async hardTest() {
+    const totalMessages = 20;
```
Comment (MaxLogvyniukFive) [inline src/modeles/movie/movie.service.ts]: Move to .env

File: src/modeles/movie/movie.service.ts [hunk 19]
```diff
@@ -49,6 +48,21 @@ export class MovieService {
...
+    for (let n = 0; n < totalMessages; n++) {
+      const datatime = new Date();
+      const miliseconds = datatime.getMilliseconds();
+      this.amqpConnection.publish(
+        'hardTestExchange',
```
Comment (MaxLogvyniukFive) [inline src/modeles/movie/movie.service.ts]: move to consts

**Review by MaxLogvyniukFive — CHANGES_REQUESTED** [review]

File: .env [hunk 20]
```diff
@@ -4,7 +4,10 @@ APP_SECRET=0dd8d1d7c673300e0e800e10e13eb6ee1414c140e046ebf7e2229010ab7ab79a10f06
 database='postgres'
```
Comment (MartinBrown-khpi) [inline .env:4]: do not store passwords/keys/secrets in the repo

File: consumer/.gitignore [hunk 21]
```diff
@@ -0,0 +1,3 @@
+dist
```
Comment (MartinBrown-khpi) [inline consumer/.gitignore:1]: .gitignore works in all subdirectories by default, isn't it?
In any case, you can divide the project into submodules
https://github.blog/2016-02-01-working-with-submodules/

File: consumer/app.ts [hunk 22]
```diff
@@ -0,0 +1,79 @@
+import amqplib from 'amqplib';
+import logger from '../winstonLogger/Logger';
+import {exchange1, hardTestExchange, publishUserInfoKey, replayKey, testKey} from '../src/common/constants';
+import {appConfig} from "./config/app.config";
+import Consul from 'consul';
+
+let consumedMessages = 0;
```
Comment (MartinBrown-khpi) [inline consumer/app.ts:7]: looks like config

**Review by MartinBrown-khpi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**